app/storybook/.storybook/blocks/SourceCode.tsx

```diff
@@ -0,0 +1,97 @@
+import { Source, useOf, Story } from '@storybook/blocks'
+import React from 'react'
+import { getStringAttributes, prettier } from '../../src/utils'
+
+type ParseStorySourceCodeProps = {
+  context: any
+  formatted?: boolean
+}
+
+/**
+ * A custom parser to get the source code of a story from the context.
+ */
+
+export const parseStorySourceCode = ({ context, formatted = false }: ParseStorySourceCodeProps) => {
+  let source: string | undefined = undefined
+
+  try {
+    const docSource = context.parameters.docs.source.originalSource
+
+    // Parse render from the source string
+    const lines = docSource.split('\n')
+    const renderRegex = /render:.*=>/
+    const renderStartIndex = lines.findIndex((line) => renderRegex.test(line))
+
+    const renderEndIndex = lines.findIndex(
+      (line, index) =>
+        index > renderStartIndex && (line === '}' || (line.substr(0, 2) === '  ' && line.charAt(2) !== ' '))
+    )
+
+    source = lines
+      .splice(renderStartIndex, renderEndIndex - renderStartIndex)
+      .join('\n')
+      .replace(renderRegex, '')
+      .replace('{...args}', getStringAttributes(context.initialArgs))
+      // Trim the leading and trailing commas
+      .replace(/,\s*$/, '')
+      .trim('')
+
+    if (formatted) {
+      source = prettier(context.parameters.codeTemplate(source, context))
+    }
+  } catch (e) {
+    console.warn(e)
+  }
+
+  return source
+}
+
+export type SourceCodeProps = {
+  of?: Record<string, any> | 'story' | 'meta'
+  transform?: (code: string, context: any) => string
+  shown?: boolean
+  hideStory?: boolean
+}
+
+/**
+ * A custom block to parse and show the source code of a story.
+ */
+
+export const SourceCode = ({ of, transform, shown = false, hideStory = false }) => {
```

Comment:
The default `Canvas` block
<img width="994" alt="image" src="https://github.com/propeldata/ui-kit/assets/55801864/935cfdf8-c39b-4273-92bd-e0429196d2b1">

It fails with parsing of `chartConfigProps` prop. Also, it adds the suffix `_stories_TimeSeries` to the component name.

Here is the output of the custom parser:
<img width="994" alt="image" src="https://github.com/propeldata/ui-kit/assets/55801864/9001edb7-a4ef-4a7b-aea0-578103f0c0ed">

Reply:
That's cool 👍 Is there any issue on the Storybook repo about adding support for this? Would be cool not to have to maintain this indefinitely.

Reply:
Well, there are a few related issues. It's not only about the invalid parsing but also about showing source code in the story mode (versus doc mode). I've already posted a note about it somewhere; to cut a long story short – due to context-related issues, the native blocks don't work properly in custom story decorators. That's why I initially came up with the custom parser. It turns out that it also parses these more complex code snippets better than the native one 🤷‍♂️